src/Routes/PrivateRoute.tsx

```diff
@@ -17,5 +17,11 @@ export const PrivateRoute = ({ hasWallet, ...rest }: PrivateRouteProps) => {
     [location],
   )
 
-  return hasWallet ? <Route {...rest} /> : <Redirect to={to} />
+  return (
+    <Switch location={location}>
+      <Route {...rest} path='/markets' />
+      <Route {...rest} path='/assets' />
```

Comment:
Any reasons why we include the assets page here? It's currently borked on asset details page with infinite loadings and weird states:

![image](https://github.com/user-attachments/assets/1f0eaf04-15c2-434f-99cf-6bfbb33267bd)

Reply:
I think it's because the details asset page is accessible from the market page but the infinit loading, copy error on the swapper makes it very weird

I would either add the login modal on click on the asset page if no wallet is connected

Or fix those states on the asset page

Reply:
To be tackled in the follow up, and the feature is under a flag